test/distributed/org/apache/cassandra/distributed/test/log/RegisterTest.java

```diff
@@ -0,0 +1,169 @@
+                                                                              true,
+                                                                              ClusterMetadataService.instance().placementProvider(),
+                                                                              LeaveStreams.Kind.UNBOOTSTRAP));
+                    UnbootstrapAndLeave unbootstrapAndLeave = (UnbootstrapAndLeave) ClusterMetadata.current().inProgressSequences.get(ClusterMetadata.current().myNodeId());
+                    ClusterMetadataService.instance().commit(unbootstrapAndLeave.startLeave);
+                    ClusterMetadataService.instance().commit(unbootstrapAndLeave.midLeave);
+                    ClusterMetadataService.instance().commit(unbootstrapAndLeave.finishLeave);
+                    ClusterMetadataService.instance().commit(new Unregister(ClusterMetadata.current().myNodeId()));
+                });
+
+                cluster.get(1).runOnInstance(() -> {
+                    ClusterMetadataService.instance().commit(SealPeriod.instance);
+                });
+
+                IInstanceConfig config = cluster.newInstanceConfig();
+                IInvokableInstance newInstance = cluster.bootstrap(config);
+                newInstance.startup();
+            }
+        }
+    }
+
+    @Test
+    public void serializationVersionDisagreementTest() throws Throwable
+    {
+        try (Cluster cluster = builder().withNodes(2)
+                                        .createWithoutStarting();
+             WithProperties prop = new WithProperties().set(TCM_ALLOW_TRANSFORMATIONS_DURING_UPGRADES, "true"))
+        {
+            cluster.get(1).startup();
+            cluster.get(1).runOnInstance(() -> {
+                try
+                {
+                    // Register a ghost node with V0 to fake-force V0 serialization. In a real world cluster we will always be upgrading from a smaller version.
+                    ClusterMetadataService.instance().commit(new Register(new NodeAddresses(InetAddressAndPort.getByName("127.0.0.10")),
+                                                                          ClusterMetadata.current().directory.location(ClusterMetadata.current().myNodeId()),
+                                                                          new NodeVersion(NodeVersion.CURRENT.cassandraVersion, Version.V0)));
+                }
+                catch (UnknownHostException e)
+                {
+                    throw new RuntimeException(e);
+                }
+                ClusterMetadataService.instance().commit(SealPeriod.instance);
+            });
+
+            cluster.get(2).runOnInstance(() -> {
+                try
+                {
+                    Field field = NodeVersion.class.getDeclaredField("CURRENT");
+                    Field modifiers = Field.class.getDeclaredField("modifiers");
```

Comment:
This will not work with JDK 17 because of JDK-8210522 (rewrite reflection of "modifiers" field)
We fixed other in-jvm tests in [CASSANDRA-18181](https://issues.apache.org/jira/browse/CASSANDRA-18181)